go/errors.md

```diff
@@ -207,3 +211,74 @@ if pkg.IsTimeout(err) {
 **Cons:**
 
 - There is a notable overhead of repeated errors to keep the contract up.
+
+## Handling Third Party Library Errors
+
+When creating libraries using third party libraries under the hood errors
+returned by them must be handled as well. Often times those third party
+libraries do not support any error handling or the provided error handling is
+not sufficient. Our way of dealing with that problem is creating internal error
+matchers which are not exposed to the library users and then using them
+transform third party errors into [custom errors](#custom-errors) which [can be
+matched](#matching-errors). Custom matchers are written in the
+`error_internal.go` file. The rest stays as usual.
+
+Example of real world `error_internal.go` file in `githubclient` package:
+
+```go
+package githubclient
+
+import "github.com/google/go-github/github"
+
+func isGitHub404(err error) bool {
```

Comment:
Would this then be a replacement for e.g. the `Prometheus` case we describe above? https://github.com/giantswarm/fmt/pull/74/files#diff-d6aadd2606fee522c4b7c216d7753d33R117